### PR TITLE
Support feature flags in guest frames

### PR DESCRIPTION
**Pull request by robertknight**
Rewrite the infrastructure that sends feature flags to annotator code, to
support feature-flagging functionality in guest frames, instead of just the host
frame.

 - Remove `FeaturesService` in the sidebar and move the logic for
   syncing feature flags from the sidebar to other frames into
   `FrameSyncService`.  This is a better place since `FrameSyncService`
   knows when frames connect and thus when to send flags to them. The
   logic for monitoring for flag changes is very simple and doesn't
   needs its own service.

 - Within `FrameSyncService`, send "featureFlagsUpdated" notifications to
   both host frames and guest frames.

 - Convert the functions and global variables in `annotator/features.js`
   into a `FeatureFlags` class, which encapsulates storage of flags,
   access to them and notifying consumers when flags change.

 - Instantiate `FeatureFlags` instances in the host frame and guest
   frame and update the flags stored in them when "featureFlagsUpdated" events
   are received from the sidebar.

### CODE REVIEW

**Comment by robertknight** [comment]
I've seen the code coverage miss. Will look at this shortly.

**Comment by lyzadanger** [comment]
> Instantiate FeatureFlags instances in the host frame and guest
frame 

I see this happening in `annotator/guest` and `annotator/sidebar`: is `sidebar` "the host frame" in this context?

**Comment by robertknight** [comment]
> I see this happening in annotator/guest and annotator/sidebar: is sidebar "the host frame" in this context?

Yes, indeed.

File: src/annotator/features.js [hunk 1]
```diff
@@ -1,31 +1,72 @@
...
+   *
+   * This will return false if the feature flags have not yet been received from
+   * the backend. Code that uses a feature flag should handle subsequent changes
+   * to the flag's state by listening for the "flagsChanged" event.
+   *
+   * @param {string} flag
+   * @return {boolean}
+   */
+  flagEnabled(flag) {
+    if (!this._knownFlags.includes(flag)) {
```
[the remark this answers is not in the document]
Comment (robertknight) [reply src/annotator/features.js:56]: Yes, indeed. Flags are a comin' in future PRs. I should have clarified that in the description.

**Pull request merged**